nextest-runner/src/runner.rs

```diff
@@ -2078,7 +2080,9 @@ where
                 self.run_stats.on_test_finished(&run_statuses);
 
                 // should this run be cancelled because of a failure?
-                let fail_cancel = self.fail_fast && !run_statuses.last_status().result.is_success();
+                let fail_cancel = self
+                    .max_fail
+                    .map_or(false, |mf| self.run_stats.failed_count() >= mf);
```

Comment:
Oh actually, one change required here -- I think there might be an off-by-one issue here.

What if `--max-fail=0` is passed in? I would expect this to mean "cancel test run after the first failure". But that seems to be `--max-fail=1`, and `--max-fail=0` would I think always trigger this condition.

Looking at [pytest](https://docs.pytest.org/en/7.1.x/how-to/failures.html#stopping-after-the-first-or-n-failures), though, it seems like they mean `--max-fail=2` to mean "cancel test run after 2 failures".

The pytest source code seems to suggest that [they treat `--max-fail=0` as `--no-fail-fast`](https://github.com/pytest-dev/pytest/blob/9d4f36d87dae9a968fb527e2cb87e8a507b0beb3/src/_pytest/main.py#L659C13-L661).

So we have a few options:

1. Do what pytest does. Remove `--max-fail=all`, use `--max-fail=0` to indicate `--no-fail-fast`, and keep everything else the same.
2. Use `>` rather than `>=` here, so shift meaning by 1.
3. Make `--max-fail=0` the same as 1.
4. Ban `--max-fail=0`, require that the minimum be 1.

Out of these, 1 and 4 are the most appealing. I think 1 makes the most sense just for similarity with pytest.